### PR TITLE
🚨🚨🚨 Update min version of accelerate to 0.26.0

### PR DESCRIPTION
**Pull request by SunMarc**
# What does this PR do ?

This PR updates the minimum version required for accelerate to `0.26.0`. 

Fixes https://github.com/huggingface/transformers/pull/31449

Creating a dev docker, waiting for the tests to pass.

### CODE REVIEW

**Review by amyeroberts — APPROVED** [review]
Thanks!

LGTM - let's have @ydshieh confirm this is all that's necessary, I'm still a bit out of touch with the official process for updating the docker images

**Review by muellerzr — APPROVED** [review]
Thanks! Agreed with @ydshieh check, though we should also have a check inside the `TrainingArguments` for this min version and raise an error in the `__post_init__`

**Comment by SunMarc** [comment]
> we should also have a check inside the TrainingArguments for this min version and raise an error in the __post_init__

Oh that's true that we don't have any check for accelerate inside TrainingArguments. Is accelerate a required dep for using `Trainer` ?

**Comment by muellerzr** [comment]
@SunMarc yes it is (the pytorch version). You can see some similar checks for `accelerate` in the training arguments. This should go in during `def _setup_devices`

**Comment by SunMarc** [comment]
> @SunMarc yes it is (the pytorch version). You can see some similar checks for accelerate in the training arguments. This should go in during def _setup_devices

Thanks ! This should be fixed in the latest commit. Also I removed a few checks that aren't needed anymore.

File: src/transformers/training_args.py [hunk 1]
```diff
@@ -1913,10 +1913,8 @@ def __post_init__(self):
             for fsdp_option in self.fsdp:
                 if fsdp_option.upper() in FSDP_SHARDING_STRATEGY:
                     # set environment variable for FSDP sharding strategy
-                    os.environ[f"{prefix}SHARDING_STRATEGY"] = (
-                        str(FSDP_SHARDING_STRATEGY.index(fsdp_option.upper()) + 1)
-                        if is_accelerate_available("0.26.0")
-                        else fsdp_option.upper()
+                    os.environ[f"{prefix}SHARDING_STRATEGY"] = str(
+                        FSDP_SHARDING_STRATEGY.index(fsdp_option.upper()) + 1
```
Comment (muellerzr) [inline src/transformers/training_args.py:1917]: Nit: I can't wait until these hacks aren't needed anymore :)

**Review by muellerzr — APPROVED** [review]
This looks good, I think the failures come from main? (they seem unrelated to me)

**Comment by SunMarc** [comment]
Failing tests are not related to my PR. See the linked PR above.

**Comment by molbap** [comment]
Failures are indeed unrelated, you can rebase on https://github.com/huggingface/transformers/pull/32649 to skip tests that are currently failing and it should work (or wait till it's merged on main)

**Review by ydshieh — APPROVED** [review]
Thanks for this update.

If IIRC, you checked the docker images for CircleCI jobs and the related tests.
Don't forget to check the images used for daily CI and the related jobs too.

**Comment by SunMarc** [comment]
> Don't forget to check the images used for daily CI and the related jobs too.

The daily CI is using the most recent version of accelerate, so that should be fine !

**Pull request merged**